### DISCUSSION TITLE
legend settings of the topology map

### DISCUSSION THREAD

**Discussion by 137-zwq** [apache/skywalking#9306] [Q&A] [answered]
After saving the legend settings of the topology map, it will expire after refreshing the page in v9, is there a problem with it?

**Comment by wu-sheng** [comment]
All settings should be saved in the storage. And all default settings are just maintained by the upstream, there is no difference. Unless you refresh the page before saving.

**Reply by 137-zwq** [reply]
But I've saved it in the database.
![数据库](https://user-images.githubusercontent.com/64053611/177281357-3f26ca67-daa4-42cf-8e19-c28ff6549cc7.jpg)
The service becomes unhealthy after the legend is set, but after the refresh, the service state becomes healthy and it still meets the unhealthy condition.
![图片](https://user-images.githubusercontent.com/64053611/177282759-56807f11-ac1e-431d-a718-c5517990f47b.jpg)

**Reply by Fine0830** [reply]
Please check your topology node data, the `isReal` needs `true`.

**Reply by 137-zwq** [reply]
The isReal is true，The topology node data is as follows：
{"data":{"topology":{"nodes":[{"id":"cWQtdWk=.1","name":"qd-ui","type":null,"isReal":true},{"id":"bG9jYWxob3N0OjMzMDY=.0","name":"localhost:3306","type":"Mysql","isReal":false},{"id":"dnVlYm9vdA==.1","name":"vueboot","type":"SpringMVC","isReal":true}],"calls":[{"id":"dnVlYm9vdA==.1-bG9jYWxob3N0OjMzMDY=.0","source":"dnVlYm9vdA==.1","detectPoints":["CLIENT"],"target":"bG9jYWxob3N0OjMzMDY=.0"},{"id":"cWQtdWk=.1-dnVlYm9vdA==.1","source":"cWQtdWk=.1","detectPoints":["SERVER"],"target":"dnVlYm9vdA==.1"}]}}}
![K0 2LR274 G3GO_NAU$AFE8](https://user-images.githubusercontent.com/64053611/177321832-7ac025d5-5fc1-4056-8da5-1e8bae9b46b8.png)
I need to click the Settings legend again for the Settings to take effect. After saving the topology Settings, the topology page is displayed again, and the service status changes to healthy.
![%IZV~TIG`)ZNQC PJMX(%F](https://user-images.githubusercontent.com/64053611/177323212-794808bc-1536-41f3-849e-327b6779e01e.png)
![@M1$%YSOFSIV%3LY$1YCB(Q](https://user-images.githubusercontent.com/64053611/177323292-06af0fa7-7537-4e8b-bcaf-ea5ea90668d7.png)

**Reply by Fine0830** [reply]
Probably you don't save your dashboard after setting legend.

**Reply by 137-zwq** [reply]
I made sure I saved the dashboard after setting up the legend. It still does not take effect after refreshing
![topo](https://user-images.githubusercontent.com/64053611/177463411-ea4305cc-2c98-495e-8a24-634d28c0be81.jpg)
![topo2](https://user-images.githubusercontent.com/64053611/177463902-3b3bdc9d-4991-4948-9e7d-a43c79f7933d.jpg)

**Comment by wu-sheng** [comment, marked as answer]
An update, @Fine0830 is checking what is going on, we could confirm that, there is an issue.

Tracked here, https://github.com/apache/skywalking/issues/9315, but GitHub doesn't show the link correctly.
